enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,384 @@
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar than OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
+is an urgent need a hotfix can be made between releases.
+
+#### Extra disk space
+Adding a plugin to `sos` and not relying on MicroShift binary means it needs
+the `oc` or `kubectl` binary installed. This package is available in the RHEL
+repos, but it comes at an extra cost of 125MB once installed.
+
+If we compare it against the minimum disk size required (which is 8GB at the
+time of this writing), it represents an extra 1.5%.
+
+### Drawbacks
+Discussed in risks and mitigations.
+
+## Design Details
+Refer to `sos` [documentation](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+for how to create plugins.
+
+The file structure should ressemble that of must-gather.
+
+### Open Questions [optional]
+N/A
+
+### Test Plan
+MicroShift only tests should be added to verify the output structure: dirs and
+files. The contents may be checked only at the file size level.
+
+An e2e test should test the command exits successfully and certain contents are
+always present.
+
+Once the plugin is introduced, it will also be used as part of the debug logs
+for all failed jobs in the MicroShift CI.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Have the plugin as part of `sos` released in RHEL9 repos.
+- QE and users feedback.
+- Initial test coverage.
+
+#### Tech Preview -> GA
+
+- User facing documentation.
+- Introduce for all other CI jobs.
+- Complete test coverage.
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+This is handled at the OS level, as `sos` is a system package.
+
+### Version Skew Strategy
+`sos` package comes with RHEL OS. We might see two different upgrade scenarios:
+1. MicroShift z-stream where the OS does not change.
+1. Any other upgrade involving the OS.
+
+In the second case, there is no issue because the OS is always upgraded first.
+In the first case, MicroShift will be at a higher version than what `sos`
+plugin was made to gather until the upgrade is finished. This should pose no
+problem unless the upgrade fails midway. If that should happen, `sos` plugin
+should be able to retrieve everything that is still compatible.
+
+This calls for having a generic approach to retrieve info from MicroShift.
+
+### Operational Aspects of API Extensions
+N/A
+
+#### Failure Modes
+N/A
+
+#### Support Procedures
+N/A
+
+## Implementation History
+N/A
+
+## Alternatives
+### Must gather
+[Must gather](https://github.com/openshift/must-gather) is the tool of choice
+for OpenShift. While this tool covers all that has been discussed in this
+enhancement, it assumes the target is cluster is a full blown OpenShift
+deployment, hence the errors we see when executing it:
+```bash
```

Comment:
I'm not sure we need the full output of the must-gather in the proposal. We know it's not working, so let's try to explain why we do not want to make it work.